### PR TITLE
Object Oriented Approach

### PR DESCRIPTION
**Pull request by dylhack**
Rather than having controllers do everything in a functional sense, each entity will manage itself making this project heavily OOP. This will allow the code to be a lot cleaner.

### CODE REVIEW

File: src/commands/categories/addcat.ts [hunk 1]
```diff
@@ -49,14 +49,16 @@ export default class AddCategory extends Command {
...
-      channelID: parent.id,
-    });
+    try {
+      await modmail.categories.create(name, emoji, parent);
+      msg.say('Category added.');
+    } catch (e) {
+      const res = 'Something internal went wrong.';
+      LogUtil.cmdError(msg, e, res);
+      msg.say(res);
+      return null;
```
Comment (DankDumpster) [inline src/commands/categories/addcat.ts:59]: Unnecessary return

File: src/database/tables/threads.ts [hunk 2]
```diff
@@ -58,44 +58,42 @@ export default class ThreadManager extends Table {
...
 
     return res.rows[0].count;
   }
 
-  /**
-   * @param {string} user
-   * @returns {Promise<Thread | null>} if thread was found
-   */
-  public async getCurrentThread(user: string): Promise<Thread | null> {
+  public async getByUser(userID: string): Promise<Thread | null> {
```
Comment (DankDumpster) [inline src/database/tables/threads.ts:70]: Comments?

**Review by DankDumpster — CHANGES_REQUESTED** [review]
And fix the mute :smile:

File: src/database/tables/threads.ts (hunk 2 above)
Comment (dylhack) [reply src/database/tables/threads.ts:70]: It's a getter

File: src/commands/categories/addcat.ts (hunk 1 above)
Comment (dylhack) [reply src/commands/categories/addcat.ts:59]: Please read the function header

Comment (dylhack) [reply src/commands/categories/addcat.ts]: Resolved in d53ec67d2ccefee67690f96695a42eee1ed82287

**Review by DankDumpster — APPROVED** [review]
Lgtm

**Pull request merged**